model/factories.go

```diff
@@ -0,0 +1,127 @@
+/*
+DESCRIPTION
+	factories.go provides a factory pattern to generate common
+	implementations of entities.
+
+AUTHORS
+	David Sutton <davidsutton@ausocean.org>
+
+LICENSE
+	Copyright (C) 2024 the Australian Ocean Lab (AusOcean).
+
+	This is free software: you can redistribute it and/or modify it
+	under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This is distributed in the hope that it will be useful, but WITHOUT
+	ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+	or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+	License for more details.
+
+	You should have received a copy of the GNU General Public License
+	in gpl.txt. If not, see http://www.gnu.org/licenses/.
+*/
+
+package model
+
+import (
+	"github.com/ausocean/utils/nmea"
+)
+
+// Sensor factories //
+
+// Consts used by sensor factories.
+const (
+	tempLinearArgs      string = "0.1,-273.15"
+	humidityScaleFactor string = "0.1"
+)
+
+// shimSensorV2 implements the fields of SensorV2 used here,
+// but replaces generic string fields with custom types.
+type shimSensorV2 struct {
```

Comment:
See also actuator PR regarding shim; I think we just need something like this
```
func sensorShim(n string, p pin, q nmea.Cod, f Func, a string, u unit, fmt Format) *SensorV2{
    return &SensorV2{n, string(p), string(q), string(f), a, string(u), string(fmt)}
}
```

Reply:
Is there any detriment to the way that I implemented it? I started implementing the way that you have suggested, but it makes the code less clear IMO. My implementation looks like
```golang
// BatteryVoltageSensor returns a default calibrated battery sensor.
func BatteryVoltageSensor(scaleFactor string) *SensorV2 {
	return shimSensorV2{
		Name:     "Battery Voltage",
		Pin:      pinBatteryVoltage,
		Quantity: nmea.DCVoltage,
		Func:     keyScale,
		Args:     scaleFactor,
		Units:    unitVoltage,
		Format:   keyRound1,
	}.SensorV2()

}
```

The shim you suggested looks like
```golang
// BatteryVoltageSensor returns a default calibrated battery sensor.
func BatteryVoltageSensor(scaleFactor string) *SensorV2 {
	return shimSensorV2(
		"Battery Voltage",
		pinBatteryVoltage,
		nmea.DCVoltage,
		keyScale,
		scaleFactor,
		unitVoltage,
		keyRound1,
	)

}
```

With my shim, it makes it clear what values are being assigned to which field.

It's slightly more code overall (an extra struct definition), but I think it might be worth it. especially because not all sensors use all the fields, and so we would be passing null values into the function.